### DISCUSSION TITLE
FBUser

### DISCUSSION THREAD

**Discussion by thanh81** [featbit/featbit#407] [Q&A] [answered]
When trying to use client.BoolVariationDetail, why can't I pass in a null for the user?  I don't have any user filters in the featureflag.  Perhaps just have the various functions overridden on the user too so we don't have to pass it in, the same as the defaultValue.

**Comment by deleteLater** [comment, marked as answer]
Hi @thanh81,

Thank you for your question. The `FbUser` parameter is required and cannot be null when we do `Variation` calls. This is because we rely on the user context to evaluate the flag value and record [insights related to the flag](https://docs.featbit.co/feature-flags/the-flag-insights).

 In your case, I would suggest you define a default user within your application and use that user in your codebase, maybe something like this:

```csharp
public static readonly FbUser FbDefaultUser = FbUser.Builder("default-user-id").Name("default").Build();
```

Or we define a built-in default user in [FbUser.cs](https://github.com/featbit/featbit-dotnet-sdk/blob/main/src/FeatBit.ServerSdk/Model/FbUser.cs), so we can use `FbUser.Default`.

```csharp
public static readonly FbUser Default = Builder("default-user-id").Name("default").Build();
```

**Reply by thanh81** [reply]
Thank you! 

That's what I ended up doing. 
I saw your post on Reddit. Timing was funny because I was just talking about your project. 

I'm seriously looking into it for our feature management solution. Just curious, why are you making it free?  It looks and works very well. 

My boss asked me if I wanted to do a show and tell with the other architects. Would you be up for doing the show and tell?

**Reply by cosmos-explorer** [reply]
Hello @thanh81 , thank you for your interest to us.

We had some extensive internal discussions about why we are making FeatBit free, here is our answer for it:

- We recognize the importance of enterprise support, including SLA (Service Level Agreement), multi-data center support etc, for numerous companies. We would charge for these valuable services.
- We understand that certain companies may require specific features that are not currently available in our solution. In such cases, we are pleased to offer paid services to facilitate the implementation of those custom requirements.

`My boss asked me if I wanted to do a show and tell with the other architects. Would you be up for doing the show and tell?`
We would love to do that. Can you give us more information about your company, the tech stack etc so we can prepare some support materials? If needed, you can join our [Slack channel](https://join.slack.com/t/featbit/shared_invite/zt-1ew5e2vbb-x6Apan1xZOaYMnFzqZkGNQ) to have more discussions

**Reply by thanh81** [reply]
We're an insurance company, primarily in the .net space.  Us working in the .net space is what attracted me to your product.  I was looking into options for feature management and ran across yours and tried it out.  

We have an architecture meeting this Thursday at 11AM EST if you'd like to join.  If so, send some email addresses and I'll set it up.  We can chat beforehand also.

**Reply by cosmos-explorer** [reply]
@thanh81 I would love to, here is my email: lian.yang.work@gmail.com
